core/src/main/java/org/apache/accumulo/core/spi/cache/BlockCacheManager.java

```diff
@@ -112,8 +112,10 @@ public interface Configuration {
    */
   public void start(Configuration conf) {
     for (CacheType type : CacheType.values()) {
-      BlockCache cache = this.createCache(conf, type);
-      this.caches.put(type, cache);
+      if (conf.getMaxSize(type) > 0) {
+        BlockCache cache = this.createCache(conf, type);
```

Comment:
Seems like this change could cause a NPE in the following code when it calls getMaxHeapSize on a cache if its null.

https://github.com/apache/accumulo/blob/bd9e67637c3ecc9e8a4e8d1ffbee26b8e7e84370/server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServerResourceManager.java#L284-L290

Reply:
Good catch, I should have looked at the references. Fixed in 72c115d.